### PR TITLE
Upgrade gradle and build

### PR DESCRIPTION
**Pull request by rmadisonhaynie**
- Updated gradle version to 5.3.1 for the java client
- Add publish method to build.gradle

### CODE REVIEW

File: build-java/resources/templates/build.gradle.mustache [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
+    id 'maven'
+    id 'java-library'
+    id 'maven-publish'
+    id 'signing'
+}
+
+group = '{{groupId}}'
+version = '{{artifactVersion}}'
+
+
```
Comment (yuancheng-p) [inline build-java/resources/templates/build.gradle.mustache]: Keep one single empty line here. :)

**Comment by yuancheng-p** [comment]
Maybe you need to specify in your commit message saying that gradle update is just for the generated Java Client, because the project itself is also in gradle, which might be confusing.

File: build-java/resources/templates/build.gradle.mustache (hunk 1 above)
Comment (rmadisonhaynie) [reply build-java/resources/templates/build.gradle.mustache]: done

**Comment by rmadisonhaynie** [comment]
> Maybe you need to specify in your commit message saying that gradle update is just for the generated Java Client, because the project itself is also in gradle, which might be confusing.

done

**Pull request merged**